### DISCUSSION TITLE
Fetching bottles for another macos version

### DISCUSSION THREAD

**Discussion by jeroen** [Homebrew/discussions#397] [Tap maintenance and Homebrew development]
I would like to use homebrew to create a redistributable bundle for an application: i.e. an archive that contains the installed formula + dependencies, but not the homebrew system itself.

For example, suppose I want `imagemagick` I could do something like:

```sh
mkdir -p bottles
HOMEBREW_CACHE="$PWD/bottles" brew fetch $(brew deps imagemagick) imagemagick
```

And then try to extract these but that is of course very clunky.

A few questions: 
 - Is is possible to `brew fetch` a bottle for another macos? Suppose I want to run the script above, but fetch all the bottles for `arm64_big_sur` while I am using catalina myself?
 - Is it possible to setup an [alternative Installation](https://github.com/Homebrew/brew/blob/master/docs/Installation.md#untar-anywhere) where the homebrew system files live in another folder than the fomula files? So that I can bootstrap a small mini homebrew intallation with only imagemagick and it's dependencies?

**Comment by SMillerDev** [comment]
> I would like to use homebrew to create a redistributable bundle for an application: i.e. an archive that contains the installed formula + dependencies, but not the homebrew system itself.

That's not a supported use of homebrew, you really should just be compiling your dependencies yourself.

---

> Is is possible to brew fetch a bottle for another macos? Suppose I want to run the script above, but fetch all the bottles for arm64_big_sur while I am using catalina myself?

Not in that way, but you could guess the bottle URLs pretty easily.

> Is it possible to setup an alternative Installation where the homebrew system files live in another folder than the fomula files? So that I can bootstrap a small mini homebrew intallation with only imagemagick and it's dependencies?

They always live in separate largely folders. I believe you can set HOMEBREW_CELLAR and HOMEBREW_REPOSITORY to override these, but I'm not sure about that.

```
> brew --repo
/usr/local/Homebrew
> brew --repo homebrew/core
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core
```

**Comment by JordanField** [comment]
Adjacent to this – I have an M1 Mac but I haven't installed `arm64` Homebrew yet, 

I'm trying to install [GLFW](https://formulae.brew.sh/formula/glfw#default) for ARM. Is it possible to use Rosetta Homebrew to download and pour the `arm64_big_sur` bottle using the non-native version?

EDIT: worked it out [possibly, now the executable I'm compiling using glfw is being killed immediately] – `arch -x86_64 brew install https://homebrew.bintray.com/bottles/glfw-3.3.2.big_sur.bottle.tar.gz`

**Reply by SMillerDev** [reply]
> I'm trying to install GLFW for ARM. Is it possible to use Rosetta Homebrew to download and pour the arm64_big_sur bottle using the non-native version?

No, because mixed dependencies make Rosetta terminate. You'll need to maintain a set of arm64 programs seperately
